tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+            port_attr = "SAI_PORT_ATTR_MULTICAST_STORM_CONTROL_POLICER_ID"
+
+        return port_attr
+
+    def check_storm_control_on_interface(self,dvs,if_name,storm_type,kbps_value):
+        print ("interface")
+        print (if_name)
+        print ("storm_type")
+        print (storm_type)
+        print ("kbps_value")
+        print (kbps_value)
+        tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+        (status,fvs) = tbl.get(if_name+"|"+storm_type)
+
+        assert status == True
+        assert len(fvs) > 0
+
+        port_oid = dvs.asicdb.portnamemap[if_name]
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("\nPort field-value pairs")
+        print (fvs)
+
+        policer_oid = 0
+
+        storm_type_port_attr = self.get_port_attr_for_storm_type(storm_type)
+
+        for fv in fvs:
+            if fv[0] == storm_type_port_attr:
+                policer_oid = fv[1]
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == True
+            print ("POlicer parameters")
+            print (fvs)
+
+        bps = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_POLICER_ATTR_CIR":
+                bps = fv[1]
+
+        #Retrieved value of bps from ASIC_DB is converted back to user input kbps
+        kbps = int(bps) / int(1000) * 8
+        print ("KBPS value")
+        print (kbps)
+
+        assert str(kbps) == str(kbps_value)
+
+
+    def add_storm_control_on_interface(self,dvs,if_name,storm_type,kbps_value):
+        print ("interface")
+        print (if_name)
+        print ("storm_type")
+        print (storm_type)
+        print ("kbps_value")
+        print (kbps_value)
+        self.add_storm_session(if_name, storm_type, kbps_value)
+        self.check_storm_control_on_interface(dvs,if_name,storm_type,kbps_value)
+
+    def test_add_storm_all_interfaces(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        tbl = swsscommon.Table(self.cdb,"PORT")
+        for key in tbl.getKeys():
+            self.add_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+
+    def test_warm_restart_all_interfaces(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        tbl = swsscommon.Table(self.cdb,"PORT")
+        for key in tbl.getKeys():
+            self.add_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.add_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+        dvs.runcmd("config save -y")
```

Comment:
please save the current config_db.json and restore it at the end of the test

Reply:
> @mohan-selvaraj, could you please clarify on the design? Why do we need to create policers for every interface and storm type vs creating different policers for different cir rates and then binding them to the port based on the storm type?

This design is based on the configuration syntax used. Industry standard configuration doesn't use a policer template and instead the policer parameters (cir/cbs,...) are configured directly 
(CLICK CLI example: "config interface storm-control broadcast add Ethernet0 10000"
IS-CLI example: "configure terminal, interface Ethernet 0, storm-control broadcast <rate/percentage/...>).
This implementation uses STORM_TYPE policer and hence takes the rate as input. 

There is no named policer in case of BUM storm-control configuration. Hence one policer is created per configuration on interface.
Policer templates can be added as an enhancement where the configuration syntax can look as - "config interface storm-control broadcast add <POLICER_TEMPLATE_NAME>.
Please suggest.